File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.c
bool judgeSquareSum(int c) {
    int r=sqrt(c);
    long s=0,j=0;
    while(s<=r)
    {
        j=s*s+r*r;
        if(j==c)
            return 1;
        if(j<c)
        s++;
        else
        r--;
    }
    return 0;

}

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long l=0,r=(int)Math.sqrt(c);
        while(l<=r)
        {
            if(l*l+r*r==c)
                return true;
            else if(l*l+r*r<c)
                l++;
            else
                r--;
        }
        return false;
    }
}